src/coreComponents/mesh/MeshBody.cpp

```diff
@@ -26,9 +27,10 @@ using namespace dataRepository;
 MeshBody::MeshBody( string const & name,
                     Group * const parent ):
   Group( name, parent ),
+  m_meshLevels( registerGroup( groupStructKeys::meshLevelsString() ) ),
   m_globalLengthScale( 0 )
 {
-  registerWrapper< integer >( viewKeys.meshLevels );
+  registerGroup< CellBlockManager >( keys::cellManager );
```

Comment:
This isn't clear to me. I interpret the relationship such that every `MeshBody` needs to have a `CellBlockManager` in order to build its base `MeshLevel`. So then it wouldn't be breaking anything if `MeshBody` knows about the existence of `CellBlockManager`.

Reply:
> `MeshBody` needs to have a `CellBlockManager` [...]

...but `MeshBody` does not have to know which implementation.
Everything that `MeshBody` needs to know is that an implementation of the abstraction `CellBlockManagerABC` will provides the meshing information. No more, no less.

- The implementation `CellBlockManager` provides more capabilities, and so it's too much for `MeshBody`.
- More important, `MeshBody` is not able to (and therefore does not have to) select which `CellBlockManager` should be used behind the scenes.
For instance, in a future refactoring we'll have an optimized `CellBlockManager` dedicated to structured `InternalMeshGenerator` (for wave propagation _e.g._), another for conformal meshes, a third one for `CornerPointGrid`...

In a nutshell, `MeshBody` uses a service (the `abstract base class`) but does not provide the service (the `implementation`).

As such, this patch would still work because we're at the beginning of the "_meshing journey_" and we have one unique implementation `CellBlockManager`.
But it hinders future refactoring actions and we'll eventually have to revert it.